modules/scratchFetch.ts

```diff
@@ -0,0 +1,218 @@
+import { defineEvent } from "strife.js";
+import fetch from "node-fetch";
+import { getSettings } from "./settings.js";
+
+defineEvent("messageCreate", async (message) => {
+	const notSet = getSettings(message.author, false)?.scratchEmbeds === undefined;
+	if (!getSettings(message.author, false).scratchEmbeds && !notSet) {
+		return;
+	}
+	const scratchUrlRegex =
+		/(?<!<)https?:\/\/scratch\.mit\.edu\/(projects|users|studios)\/\w+\/?(?!>)/; //gpt wrote the regex and like half of this code
+	const match = message.content.match(scratchUrlRegex);
+
+	if (!match) {
+		return;
+	}
+
+	const urlParts = match[0].split("/");
+	const type = urlParts[3]; // m ybrain is dead
+
+	async function fetchApiData(apiUrl: string): Promise<any> {
+		try {
+			const response = await fetch(apiUrl);
+
+			const data = await response.json();
+			return data;
+		} catch (error) {}
+	}
+	function long(text: string, maxLength: number, appendText: string): string {
```

Comment:
i believe a `truncateText` function already exists. you'll need to `escapeMessage` it first tho